agreements/notePurchaseAgreementsService.go

```diff
@@ -77,25 +97,30 @@ func (service *NotePurchaseAgreementService) Save(ctx context.Context, agreement
 		HomeAddress:    agreement.GetHomeAddress(),
 		PhoneNumber:    agreement.GetPhoneNumber(),
 		SocialSecurity: agreement.GetSocialSecurity(),
-		CreatedOn:      agreement.GetCreatedOn(),
+		CreatedOn:      time.Now().Format(time.RFC3339),
 	}
```

Comment:
This belongs in a gateway.  The purpose of the gateways is to isolate any data specific code from the domain layer.  I'm concerned that if we don't keep that separation we'll have problems with "spaghetti code" in the future.  This separation of the data layer from the domain layer will help us avoid major maintenance issues in the future.

Reply:
Hi Scott, are you referring to the CreatedOn field? I can move that to the gateway layer. Do you have a preference for the creation of ids for the user and NPA? Ideally, we can also move that to the gateway layer.

Reply:
Updated using the suggestion.

Reply:
I'm referring to contaminating the service layer with the data package.  I would instantiate data.User and any other data structs into the gateway layer.  Personally, I would use a domain model which would interact with an interface to the gateway, but if you want to avoid that complexity, I suggest keeping all data package references in the gateway code and out of the service code.

Reply:
for CreatedOn, This is actually a field that the database can calculate.  I think it's a good idea to let the database or the ORM handle things like "CreatedOn", "UpdatedOn".